### ISSUE TITLE
RFC: Rename mastertrainer?

### ISSUE THREAD

**Issue by stweil** [tesseract-ocr/tesseract#3569]
Should we rename `mastertrainer` to `maintrainer` (maybe easily confused with `maintainer`) or perhaps `tesstrainer` for 5.0.0?

As such a renaming is incompatible with previous releases, it requires a new major version, so if we want to do it, we have to do it soon (for 5.0.0) or much later (for 6.0.0, 7.0.0, ...).

**Comment by stweil** [comment]
Related: issue #3554.

**Comment by amitdo** [comment]
`mastertrainer.cpp|h` -> `legacytrainerutils.cpp|h`

`MasterTrainer` -> `LegacyTrainer` 

https://github.com/tesseract-ocr/tesseract/search?q=MasterTrainer

I suggest to wait a few days for more feedback.

**Comment by amitdo** [comment]
BTW, what about `blacklist` and `whitelist`? See here[](https://www.google.com/search?q=%22blacklist%22+%22whitelist%22+%22offensive%22).

**Comment by stweil** [comment]
> BTW, what about `blacklist` and `whitelist`?

Yes, those would need a change, too: `tessedit_char_blacklist`, `tessedit_char_unblacklist`, `tessedit_char_whitelist`.

Filenames and class names are not part of the external interface, so we can fix those any time later.
